PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env python
+"""
+This code is for downloading course information from De Anza myportal.
+
+It requires the users input their own user name and password.
+"""
+
+from selenium import webdriver
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import Select
+from selenium.common.exceptions import NoSuchElementException
+import sys
+import csv
+from bs4 import BeautifulSoup
+import time
+import chromedriver_binary
+import logging
+
+
+def submitClick(driver):
+    """
+    Search and click sumbit button.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    ipts = driver.find_elements_by_tag_name("input")
+    submit = ""
+    for ipt in ipts:
+        if "submit".lower() == ipt.get_attribute("type") and ipt.is_enabled() and ipt.is_displayed():
+            submit = ipt
+            break
+    if not submit:
+        logging.info("No submit button found！")
+        return
+    submit.click()
+    time.sleep(2)  # Waiting for completion
+
+
+def advanceSearch(driver):
+    """
+    Locate submit button and click.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    # Locate and click the Submit button
+    ipts = driver.find_elements_by_tag_name("input")
+    submit = ""
+    for ipt in ipts:
+        if "submit".lower() == ipt.get_attribute("type") and \
+                "Advanced Search" == ipt.get_attribute("value"):
+            submit = ipt
+            break
+    if not submit:
+        logging.info("Advanced Search button not found！")
+        return
+    submit.click()
+
+
+# login
+def login_myportal(driver):
+    """
+    Open myportal website and login.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    # Open the browser (The true url is：https://myportal.fhda.edu/）
+    driver.get("https://myportal.fhda.edu/")
+    # input username
+    username = int(input("Please enter your user name:  "))
+    driver.find_element_by_id("j_username").send_keys(username)
+    # input password
+    password = input("Please enter your password:   ")
+    driver.find_element_by_id("j_password").send_keys(password)
+    driver.find_element_by_id(
+        "btn-eventId-proceed").click()  # Wait for the response from the next page and make sure the page is loaded!
+    logging.info("Log in finished.")
+
+
+# Open course search page
+def openSearchPage(driver):
+    """
+    Open search page.
+
+    Click Appls->Look Up Classes-> open search page
+    Parameters: webdriver
+    Returns: None
+    """
+    # Find the Apps menu and click
+    findAppsMenu(driver)
+    # Find in App List（Look Up Classes）button
+    classes = lookUpClasses(driver)
+    mainWindowName = driver.window_handles[0]  # Main window name
+
+    classes.click()
```

Comment:
Recall that in your previous function, if the class is not found, you return nothing:
```
    if not classes:
        logging.info("No Look Up Classes feature found in the app list!")
        return
```
So you can not simply do `classes.click()` here.

Reply:
Thank you! I changed to sys.exit(-1)